tls/extensions/s2n_client_key_share.c

```diff
@@ -56,30 +60,51 @@ const s2n_extension_type s2n_client_key_share_extension = {
     .if_missing = s2n_extension_noop_if_missing,
 };
 
-static int s2n_ecdhe_supported_curves_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+static int s2n_generate_preferred_key_shares(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
     notnull_check(conn);
-    notnull_check(conn->config);
+
+    uint8_t preferred_key_shares = conn->preferred_key_shares;
+    const struct s2n_ecc_named_curve *named_curve = NULL;
+    struct s2n_ecc_evp_params *ecc_evp_params = NULL;
 
     const struct s2n_ecc_preferences *ecc_pref = NULL;
     GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
     notnull_check(ecc_pref);
 
-    const struct s2n_ecc_named_curve *named_curve = NULL;
-    struct s2n_ecc_evp_params *ecc_evp_params = NULL;
+    /* If lsb is set, skip keyshare generation for all curve */
+    if (S2N_GENERATE_EMPTY_KEY_SHARE_LIST(preferred_key_shares)) {
+        GUARD(s2n_stuffer_write_uint16(out, 0));
+        return S2N_SUCCESS;
+    }
 
-    for (uint32_t i = 0; i < ecc_pref->count; i++) {
+    if (!conn->preferred_key_shares) {
```

Comment:
won't this be always true ?

Reply:
No, it won't be true always. It will be true only for the default case i.e when an override is not requested by calling `s2n_connection_set_keyshare_by_group_for_testing` API.